front/src/modules/conflict/utils.ts

```diff
@@ -102,6 +102,29 @@ export default function addTrainNamesToConflicts(
   });
 }
 
+export const reorderConflictTrains = (
+  trainsData: ConflictWithTrainNames['trainsData'],
+  selectedTrainName?: string | null
+): ConflictWithTrainNames['trainsData'] => {
+  if (!trainsData.length) return trainsData;
+  if (!selectedTrainName) {
+    return [...trainsData].sort((a, b) => a.name.length - b.name.length);
+  }
+
+  // Find the selected train
+  const selectedTrainIndex = trainsData.findIndex((train) => train.name === selectedTrainName);
+  if (selectedTrainIndex <= 0) {
```

Comment:
I think we should do that, otherwise the selected train won't always appear first (and change the below comment)
```suggestion
  if (selectedTrainIndex < 0) {
```

Reply:
Done